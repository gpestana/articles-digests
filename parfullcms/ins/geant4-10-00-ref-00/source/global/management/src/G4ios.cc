//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
//
// $Id: G4ios.cc 70021 2013-05-22 07:55:29Z gcosmo $
//
// 
// --------------------------------------------------------------
//      GEANT 4 class implementation file
//
// G4ios.cc
//
// History 1998 Nov. 3 Masayasu Nagamatu

#include "G4ios.hh"
#include "G4strstreambuf.hh"

#ifdef G4MULTITHREADED
  G4ThreadLocal G4strstreambuf *G4coutbuf_p = 0;
  G4ThreadLocal G4strstreambuf *G4cerrbuf_p = 0;
  G4ThreadLocal std::ostream *G4cout_p = 0;
  G4ThreadLocal std::ostream *G4cerr_p = 0;
  #define G4coutbuf (*G4coutbuf_p)
  #define G4cerrbuf (*G4cerrbuf_p)
  #define G4cout (*G4cout_p)
  #define G4cerr (*G4cerr_p)

  void G4iosInitialization()
  {
    if (G4coutbuf_p == 0) G4coutbuf_p = new G4strstreambuf;
    if (G4cerrbuf_p == 0) G4cerrbuf_p = new G4strstreambuf;
    if (G4cout_p == 0) G4cout_p = new std::ostream(G4coutbuf_p);
    if (G4cerr_p == 0) G4cerr_p = new std::ostream(G4cerrbuf_p);
  }

  void G4iosFinalization()
  {
    delete G4cout_p;
    delete G4cerr_p; 
    delete G4coutbuf_p;
    delete G4cerrbuf_p;
  }

#else  // Sequential

  G4strstreambuf G4coutbuf;
  G4strstreambuf G4cerrbuf;
  std::ostream G4cout(&G4coutbuf);
  std::ostream G4cerr(&G4cerrbuf);

  void G4iosInitialization() {}
  void G4iosFinalization() {}

#endif
