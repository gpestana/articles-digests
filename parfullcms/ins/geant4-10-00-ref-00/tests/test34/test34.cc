//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//

#include <algorithm>
#include <iostream>
#include "G4Timer.hh"

// G4 includes 
#include "G4ios.hh"
#include "G4GlobalFastSimulationManager.hh"
#include "G4Timer.hh"
#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4UItcsh.hh"

// my project 
#include "Tst34DetectorConstruction.hh"
#include "Tst34PhysicsList.hh"
#include "Tst34PrimaryGeneratorAction.hh"
#include "Tst34EventAction.hh"
#include "Tst34RunAction.hh"

G4Timer Timer;
G4Timer Timerintern;

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......

int main(int argc,char** argv)
{
  // Timer to see GFlash performance
  Timer.Start();

  G4cout<<"+-------------------------------------------------------+"<<G4endl;
  G4cout<<"|                                                       |"<<G4endl;
  G4cout<<"|          This is an example of Shower                 |"<<G4endl;
  G4cout<<"|          Parameterization with GFLASH                 |"<<G4endl;
  G4cout<<"+-------------------------------------------------------+"<<G4endl;

  G4RunManager* runManager = new G4RunManager;

  // UserInitialization classes (mandatory)
  G4cout<<"# GFlash Example: Detector Construction"<<G4endl;    
  runManager->SetUserInitialization(new Tst34DetectorConstruction);
  G4cout<<"# GFlash Example: Physics list"<<G4endl;
  runManager->SetUserInitialization(new Tst34PhysicsList);
  G4cout<<"# GFlash Example: Primary Generator"<<G4endl;
  runManager->SetUserAction(new Tst34PrimaryGeneratorAction);
  G4cout<<"# GFlash Example: User Action Classes"<<G4endl;
  runManager->SetUserAction(new Tst34EventAction);
  runManager->SetUserAction(new Tst34RunAction);

  G4UImanager* UI = G4UImanager::GetUIpointer();
  UI->ApplyCommand("/run/verbose 0");
  runManager->Initialize();
  UI->ApplyCommand("/Step/Verbose 0");

  if (argc==1)   // Define UI terminal for interactive mode  
  { 
    // G4UIterminal is a (dumb) terminal.
    G4UIsession * session = 0;
#ifdef G4UI_USE_TCSH
      session = new G4UIterminal(new G4UItcsh);      
#else
      session = new G4UIterminal();
#endif    
    session->SessionStart();
    delete session;
  }
  else           // Batch mode
  { 
    G4String s=*(argv+1);
    UI->ApplyCommand("/control/execute "+s);
  }

  delete runManager;

  Timer.Stop();
  G4cout << G4endl;
  G4cout << "******************************************";
  G4cout << G4endl;
  G4cout << "Total Real Elapsed Time is: "<< Timer.GetRealElapsed();
  G4cout << G4endl;
  G4cout << "Total System Elapsed Time: " << Timer.GetSystemElapsed();
  G4cout << G4endl;
  G4cout << "Total GetUserElapsed Time: " << Timer.GetUserElapsed();
  G4cout << G4endl;
  G4cout << "******************************************";
  G4cout << G4endl;

  return 0;
}

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
