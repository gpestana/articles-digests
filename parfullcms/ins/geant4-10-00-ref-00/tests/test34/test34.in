#
# Macro file for the initialization phase of "test34"
#
# Sets some default verbose
#

/gps/pos/centre 0 0 0
/gps/particle  e-
/gps/energy   20 GeV
/gps/direction  0. 0. 1.0

# Paramatrisation off
/GFlash/flag 1

/tracking/verbose 0

/run/beamOn 10
exit
