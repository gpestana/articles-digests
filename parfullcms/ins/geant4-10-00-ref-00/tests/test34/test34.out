+-------------------------------------------------------+
|                                                       |
|          This is an example of Shower                 |
|          Parameterization with GFLASH                 |
+-------------------------------------------------------+

        ############################################
        !!! WARNING - FPE detection is activated !!!
        ############################################

*************************************************************
 Geant4 version Name: geant4-10-00-ref-00    (6-December-2013)
                      Copyright : Geant4 Collaboration
                      Reference : NIM A 506 (2003), 250-303
                            WWW : http://cern.ch/geant4
*************************************************************

# GFlash Example: Detector Construction
Tst34DetectorConstruction::Detector constructor
# GFlash Example: Physics list
start construct particle
end construct particle
# GFlash Example: Primary Generator
# GFlash Example: User Action Classes
Defining the materials
Shower parameterization
/********************************************/ 
  - GVFlashShowerParameterisation::Material -  
        Material : PbWO4
   Z = 68.3599
   A = 170.87
   X0 = 0.892421 cm
    Rm= 1.73752 cm
   Ec = 10.786 MeV
/********************************************/ 
/********************************************/ 
  - GFlashHomoShowerParameterisation::Constructor -  
/********************************************/ 
end shower parameterization
FastSimulationManagerProcess
AddParameterisation
Physics List constructor
ConstructEM

conv:   for  gamma    SubType= 14
      Lambda table from 1.022 MeV to 10 TeV in 49 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        BetheHeitler :  Emin=        0 eV    Emax=       80 GeV
     BetheHeitlerLPM :  Emin=       80 GeV   Emax=       10 TeV

compt:   for  gamma    SubType= 13
      Lambda table from 100 eV  to 1 MeV in 28 bins, spline: 1
      LambdaPrime table from 1 MeV to 10 TeV in 49 bins 
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Klein-Nishina :  Emin=        0 eV    Emax=       10 TeV

phot:   for  gamma    SubType= 12
      LambdaPrime table from 200 keV to 10 TeV in 54 bins 
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       PhotoElectric :  Emin=        0 eV    Emax=       10 TeV   AngularGenSauterGavrila

msc:   for e-    SubType= 10
      RangeFactor= 0.04, stepLimitType: 1, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

eIoni:   for  e-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :  Emin=        0 eV    Emax=       10 TeV

eBrem:   for  e-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
             eBremSB :  Emin=        0 eV    Emax=        1 GeV   DipBustGen
            eBremLPM :  Emin=        1 GeV   Emax=       10 TeV   DipBustGen

msc:   for e+    SubType= 10
      RangeFactor= 0.04, stepLimitType: 1, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

eIoni:   for  e+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :  Emin=        0 eV    Emax=       10 TeV

eBrem:   for  e+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
             eBremSB :  Emin=        0 eV    Emax=        1 GeV   DipBustGen
            eBremLPM :  Emin=        1 GeV   Emax=       10 TeV   DipBustGen

annihil:   for  e+    SubType= 5
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            eplus2gg :  Emin=        0 eV    Emax=       10 TeV

hIoni:   for  proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=        2 MeV
          BetheBloch :  Emin=        2 MeV   Emax=       10 TeV

msc:   for proton    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

hIoni:   for  GenericIon    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=        2 MeV
          BetheBloch :  Emin=        2 MeV   Emax=       10 TeV

msc:   for GenericIon    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV

hIoni:   for  alpha    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=   7.9452 MeV
          BetheBloch :  Emin=   7.9452 MeV   Emax=       10 TeV

msc:   for alpha    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

hIoni:   for  anti_proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :  Emin=        0 eV    Emax=        2 MeV
          BetheBloch :  Emin=        2 MeV   Emax=       10 TeV

msc:   for anti_proton    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

hIoni:   for  kaon+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=  1.05231 MeV
          BetheBloch :  Emin=  1.05231 MeV   Emax=       10 TeV

msc:   for kaon+    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

hIoni:   for  kaon-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :  Emin=        0 eV    Emax=  1.05231 MeV
          BetheBloch :  Emin=  1.05231 MeV   Emax=       10 TeV

msc:   for kaon-    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

muIoni:   for  mu+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=      200 keV
          BetheBloch :  Emin=      200 keV   Emax=        1 GeV
        MuBetheBloch :  Emin=        1 GeV   Emax=       10 TeV

msc:   for mu+    SubType= 10
      RangeFactor= 0.2, step limit type: 0, lateralDisplacement: 1, polarAngleLimit(deg)= 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

muBrems:   for  mu+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :  Emin=        0 eV    Emax=       10 TeV

muPairProd:   for  mu+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      Sampling table 17x1001 from 1 GeV to 10 TeV 
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :  Emin=        0 eV    Emax=       10 TeV

muIoni:   for  mu-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :  Emin=        0 eV    Emax=      200 keV
          BetheBloch :  Emin=      200 keV   Emax=        1 GeV
        MuBetheBloch :  Emin=        1 GeV   Emax=       10 TeV

msc:   for mu-    SubType= 10
      RangeFactor= 0.2, step limit type: 0, lateralDisplacement: 1, polarAngleLimit(deg)= 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

muBrems:   for  mu-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :  Emin=        0 eV    Emax=       10 TeV

muPairProd:   for  mu-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      Sampling table 17x1001 from 1 GeV to 10 TeV 
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :  Emin=        0 eV    Emax=       10 TeV

hIoni:   for  pi+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :  Emin=        0 eV    Emax=  297.505 keV
          BetheBloch :  Emin=  297.505 keV   Emax=       10 TeV

msc:   for pi+    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

hIoni:   for  pi-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :  Emin=        0 eV    Emax=  297.505 keV
          BetheBloch :  Emin=  297.505 keV   Emax=       10 TeV

msc:   for pi-    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            UrbanMsc :  Emin=        0 eV    Emax=       10 TeV  Table with 77 bins Emin=    100 eV    Emax=     10 TeV

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes
 Material : Air
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  990 eV     e-  990 eV     e+  990 eV  proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes
 Material : PbWO4
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  89.3347 keV    e-  1.12535 MeV    e+  1.05931 MeV proton 100 keV
 Region(s) which use this couple : 
    crystals

====================================================================

### Run 0 start.
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 0


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0.01
******************************************
 ------ Tst34EventAction::End of event nr. 0  -----
  5341 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9452 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 1


******************************************
Internal Real Elapsed Time is: 0.01
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 1  -----
  5326 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9073 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 2


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 2  -----
  5306 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9687 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 3


******************************************
Internal Real Elapsed Time is: 0.01
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0.01
******************************************
 ------ Tst34EventAction::End of event nr. 3  -----
  5869 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9535 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 4


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 4  -----
  5335 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9713 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 5


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 5  -----
  5311 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9422 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 6


******************************************
Internal Real Elapsed Time is: 0.01
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0.01
******************************************
 ------ Tst34EventAction::End of event nr. 6  -----
  5323 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9574 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 7


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 7  -----
  5322 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9568 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 8


******************************************
Internal Real Elapsed Time is: 0
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 8  -----
  5321 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9656 (GeV)
::Initializing the sensitive detector
 ------ Start Tst34EventAction ----- 
 Start generating event Nr 9


******************************************
Internal Real Elapsed Time is: 0.01
Internal System Elapsed Time: 0
Internal GetUserElapsed Time: 0
******************************************
 ------ Tst34EventAction::End of event nr. 9  -----
  5321 hits are stored in Tst34HitsCollection 
 #### Tst34EventAction::Test: Total energy deposited in the calorimeter: 19.9511 (GeV)
number of events = 10

******************************************
Total Real Elapsed Time is: 0.84
Total System Elapsed Time: 0.01
Total GetUserElapsed Time: 0.74
******************************************
